test-suite-csrf/src/test/java/io/micronaut/views/tests/security/csrf/CsrfSessionLogingHandlerTest.java

```diff
@@ -0,0 +1,82 @@
+package io.micronaut.views.tests.security.csrf;
+
+import io.micronaut.context.annotation.Property;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.http.*;
+import io.micronaut.http.annotation.*;
+import io.micronaut.http.client.BlockingHttpClient;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.micronaut.http.cookie.Cookie;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.authentication.AuthenticationRequest;
+import io.micronaut.security.authentication.AuthenticationResponse;
+import io.micronaut.security.authentication.provider.HttpRequestAuthenticationProvider;
+import io.micronaut.security.csrf.repository.CsrfTokenRepository;
+import io.micronaut.security.rules.SecurityRule;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import io.micronaut.views.View;
+import jakarta.inject.Singleton;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@Property(name = "micronaut.security.authentication", value = "session")
+@Property(name = "micronaut.security.redirect.enabled", value = StringUtils.FALSE)
+@Property(name = "micronaut.security.csrf.filter.regex-pattern", value = "^(?!\\/login).*$")
+@Property(name = "micronaut.security.csrf.signature-key", value = "pleaseChangeThisSecretForANewOnekoqQ-EstJQLr_T-1qS0gZH75aKtMN3Yj0iPS4hcgUuTwjAzZr1Z9CAow")
+@Property(name = "spec.name", value = "CsrfSessionLogingHandlerTest")
+@MicronautTest
+class CsrfSessionLogingHandlerTest {
+
+    @Test
+    void loginSavesACsrfTokenInSession(@Client("/") HttpClient httpClient, CsrfTokenRepository<HttpRequest<?>> csrfTokenRepository) {
```

Comment:
this tests the successful case, need a test for the failure case

Reply:
resolved via https://github.com/micronaut-projects/micronaut-views/pull/914/commits/92abf2e0e1583891129602e85677f5fcfcf86c44